### ISSUE TITLE
useSelector not memoized

### ISSUE THREAD

**Issue by lishine** [LegendApp/legend-state#267]
> `useSelector` computes a value and automatically listens to any observables accessed while running, and only re-renders if the computed value changes.

`useSelector` rerendered the component  despite the value `pluginCount` was not changed.
The component is wrapped in `observer` and also this used `enableReactTracking({ auto: true })`

```ts
const pluginsCount = useSelector(() => {
    return items$.get().filter((v) => !v.state.hide).length
})
```

So I wrapped it in `computed` for the meantime:

```ts
const pluginsCount = useSelector(() => {
    return computed(() => {
        return items$.get().filter((v) => !v.state.hide).length
    }).get()
})
```

**Comment by lishine** [comment]
My ideal `useSelector` is:
1. Memoized
2. Recreated upon `deps` change
Is a combination of the current `useComputed` implementation wrapped with `useSelector`

**Comment by lishine** [comment]
`useObserve` is also missing deps

**Comment by lishine** [comment]
`proxy()` to stringify an object if received

**Comment by jmeistrich** [comment]
That's odd, it should not be re-rendering if the return value of useSelector is the same. What version are you using? Can you share a sandbox or repo that shows that happening? 

I haven't seen that behavior and we even have a test to ensure it renders only once: https://github.com/LegendApp/legend-state/blob/900f02f9a42d3aa54cbf18feaadc7c13e8d5c96e/tests/react.test.ts#L307

**Comment by lishine** [comment]
I checked in codesandbox and the problem reproduced. 
Look at the html output (in order to see the correct console logs run it in separate window, codesandbox sucks) 
https://codesandbox.io/p/sandbox/legend-app-dqygkd?file=%2Fsrc%2FApp.tsx
Your test indeed runs ok.
And I added similar test to codesandbox and did not reproduce as well

```ts
    test('2 useSelector renders once when it returns the same thing', () => {
        const obs = observable(0);
        let num = 0;
        let num2 = 0;
        let lastValue = 0;

        const Test = function Test() {
            num2++;
            lastValue = useSelector(() => {
                num++;
                return obs.get() === 0 ? 0 : 1;
            });

            return createElement('div', undefined);
        };
        function App() {
            return createElement(Test);
        }
        render(createElement(App));

        expect(lastValue).toEqual(0);
        expect(num).toEqual(1);
        expect(num2).toEqual(1);
        act(() => {
            obs.set((s) => s + 1);
        });
        expect(num).toEqual(3);
        expect(num2).toEqual(2);
        act(() => {
            obs.set((s) => s + 1);
        });
        expect(num).toEqual(4);
        expect(num2).toEqual(2);
    });
```

Don't know what is happening. But
It has to be memoized.
It has to use deps instead of running the callback on each render.

**Comment by jmeistrich** [comment]
Actually nevermind, I see what you mean and I made a failing test for it. I'm looking into it.

**Comment by jmeistrich** [comment]
Fixed in 2.1.12. Thanks for catching that!
